app/com/gu/floodgate/DynamoDBTable.scala

```diff
@@ -0,0 +1,91 @@
+package com.gu.floodgate
+
+import com.amazonaws.services.dynamodbv2.AmazonDynamoDBAsync
+import com.amazonaws.services.dynamodbv2.model._
+import com.amazonaws.handlers.AsyncHandler
+import com.typesafe.scalalogging.StrictLogging
+import scala.collection.JavaConversions._
+import scala.collection.JavaConverters._
+import scala.concurrent.{ Promise, Future }
+
+trait DynamoDBTable[T] extends StrictLogging {
+
+  protected val dynamoDB: AmazonDynamoDBAsync
+  protected val tableName: String
+  protected def fromItem(item: Map[String, AttributeValue]): T
+  protected def toItem(t: T): Map[String, AttributeValue]
+  protected def toItemUpdate(t: T): Map[String, AttributeValueUpdate]
+
+  def getAll(): Future[Seq[T]] = {
+    val request = new ScanRequest().withTableName(tableName)
```

Comment:
Yes, from what I could discern it doesn't support async operations. Would be good to know if there's a better way of doing this kind of thing.